### PR TITLE
139 ezb enrichment

### PR DESCRIPTION
**Pull request by philboeselager**

### CODE REVIEW

**Review by hornmo — COMMENTED** [review]
This raises some questions we have to discuss, mainly if we even want to create ReviewRequests for EZB conflicts.

The changes in TippService seem out of place and don't really make sense to me.

File: server/gokbg3/grails-app/services/org/gokb/TitleAugmentService.groovy [hunk 1]
```diff
@@ -203,12 +205,106 @@ class TitleAugmentService {
     }
   }
 
+  def augmentEzb(titleInstance) {
+    log.debug("Augment EZB - TitleInstance: ${titleInstance.niceName} - ${titleInstance.class?.name}")
+    CuratoryGroup editorialGroup = grailsApplication.config.gokb.ezbAugment?.rrCurators ?
+        CuratoryGroup.findByNameIlike(grailsApplication.config.gokb.ezbAugment.rrCurators) : null
```
Comment (hornmo) [inline server/gokbg3/grails-app/services/org/gokb/TitleAugmentService.groovy]: if augment is activated via config, we should probably create the `CuratoryGroup` here if it doesn't exist.

File: server/gokbg3/grails-app/init/gokbg3/BootStrap.groovy [hunk 2]
```diff
@@ -235,7 +236,9 @@ class BootStrap {
         log.debug("Checking for missing component statistics")
         ComponentStatisticService.updateCompStats()
 
-        log.info("GoKB Init complete");
+        AugmentJob.schedule(grailsApplication.config.gokb.augment.cron)
```
Comment (hornmo) [inline server/gokbg3/grails-app/init/gokbg3/BootStrap.groovy]: This does not seem to work right now, and I cannot find the corresponding method.

File: server/gokbg3/grails-app/services/org/gokb/TitleAugmentService.groovy [hunk 3]
```diff
@@ -203,12 +205,106 @@ class TitleAugmentService {
...
+      }
+      else {
+        log.debug("Skipping title with existing RR ..")
+      }
+    }
+  }
+
   private void setNewTitleInfo(titleInstance, info) {
-    if (!titleInstance.publishedFrom && info.publishedFrom) {
+    if ((!titleInstance.hasProperty('publishedFrom') || !titleInstance.publishedFrom) && info.publishedFrom) {
```
Comment (hornmo) [inline server/gokbg3/grails-app/services/org/gokb/TitleAugmentService.groovy]: The property is defined in the generic base class `TitleInstance`, therefore this check shouldn't be necessary.

**Review by hornmo — CHANGES_REQUESTED** [review]
The changes in `TippService` are overhauled and should be removed from this PR.

**Review by philboeselager — COMMENTED** [review]
can't be approved by the raiser of the PR. Feel dree, @hornmo :-)

**Review by hornmo — APPROVED** [review]

**Pull request merged**